### DISCUSSION TITLE
Supabase: Failed to toggle RLS: failed to update pg.tables with the given ID: Connection terminated due to connection timeout

### DISCUSSION THREAD

**Discussion by CybertronianKelvin** [prisma/prisma#21416] [Q&A] [answered]
### Question

Has anyone worked with Prisma and Supabase lately and enabled Row-level Security on Supabase? I keep getting the error 

```Failed to toggle RLS: failed to update pg.tables with the given ID: Connection terminated due to connection timeout```

However, I can't write a schema in my Prisma file with the id. From the error, I reckon this is what superbase expects in the schema:

```prisma
model test_table {
  text   String
  userId String? @db.Uuid
  user   users?  @relation(fields: [userId], references: [id])
  @@index([userId])
  @@schema("public")
}
```

But get screamed by Prisma that the table needs an idea. Any suggestions or directions will be much appreciated, thanks guys

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @CybertronianKelvin 👋 

Thank you for raising this question.

The error message you're getting is telling you that the connection to the database timed out while Prisma was trying to toggle RLS. 

Prisma is expecting you to specify the `ID` of the table in the schema, but you're not doing that. To fix this, you need to add the `ID` field to your test_table model.

Here's an example:
```prisma
model test_table {
  id      Int   @id @default(autoincrement())
  text   String
  userId String? @db.Uuid
  user   users?  @relation(fields: [userId], references: [id])
  @@index([userId])
  @@schema("public")
}
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
